pyproject.toml

```diff
@@ -6,3 +6,7 @@ target-version = ['py37']
 profile = "black"
 multi_line_output = 3
 py_version = "37"
+
+[tool.pyright]
+pythonVersion = "3.7"
+reportGeneralTypeIssues = false
```

Comment:
I'm pretty sure we should be able to fix these. I'd be very surprised if these were spurious.

Reply:
While the ones about shadowing variables are easy to fix (although I don't really see the necessity), the last one about 
```
  /home/mkosi/mkosi/backend.py:107:15 - error: Cannot assign member "package_type" for type "Distribution"
  Member "package_type" is unknown (reportGeneralTypeIssues)
 ```
really stumps me. I think that's a point where Python is just too dynamic for static analysis.

Reply:
OK, I actually missed that, it's kinda understandable that typing doesn't work there. Can we add `package_type: PackageType` to the enum declaration to fix this?

Reply:
I don't really know how. I mean, we're doing that right now in the canonical way. I'm not sure I'd call this a bug in pyright, but it's a missing feature and something that they should specialcase. For now I'd say there's two ways to get rid of the error: Switch of `reportGeneralTypeIssues` for pyright or adding a `# type: ignore` which blinds both pyright and mypy to this line. I'd lean to the former.

Reply:
I mean something like this:

```
class Distribution(enum.Enum):
    package_type: PackageType
    fedora = 0, PackageType.rpm
    ....
```

Reply:
Ah, that does work!